float64 on
float64 obX
float64 obY
float64 trX
float64 trY



